tests/e2e_vaultwarden_test.py

```
@@ -1,26 +1,60 @@
+    timeout: float = 300,
+) -> str:
+    # Redact passwords from log output
+    safe_cmd = " ".join(
+        "***"
+        if i > 0
+        and command[i - 1]
+        in ("--keepass-password", "--bitwarden-password", "--passwordenv")
+        else arg
+        for i, arg in enumerate(command)
     )
+    logger.info(f"Running: {safe_cmd}")
```

Ruff (0.15.2)

[warning] Logging statement uses f-string

(G004)